File: ProblemSet/Week 2/Problem/JavaScript/Queue.js
let queue = [];

/**
 * Displays the queue menu to the user
 */
function displayMenu() {
  console.log("Queue Menu:");
  console.log("1. Add elements to the queue");
  console.log("2. Remove an element from the front of the queue");
  console.log("3. Add another element to the queue");
  console.log("4. Exit");
}

/**
 * Adds elements to the queue
 */
function addElementsToQueue() {
  // Ask the user to enter the number of elements to add to the queue
  let numElements = parseInt(prompt("Enter the number of elements to add to the queue: "));

  // Add elements to the queue
  for (let i = 0; i < numElements; i++) {
    let element = prompt(`Element ${i + 1}: `);
    queue.push(element);
  }

  // Print the queue elements
  console.log("Queue elements:");
  console.log(queue);
}

/**
 * Removes an element from the front of the queue
 */
function removeElementFromFront() {
  // Check if the queue is not empty
  if (queue.length > 0) {
    // Remove an element from the front of the queue
    let removedElement = queue.shift();
    console.log("Removed element: " + removedElement);
    console.log("Updated queue elements:");
    console.log(queue);
  } else {
    console.log("Queue is empty. Cannot remove an element.");
  }
}

/**
 * Adds another element to the queue
 */
function addAnotherElementToQueue() {
  // Ask the user to enter the new element
  let newElement = prompt("Enter the new element: ");
  queue.push(newElement);
  console.log("Updated queue elements:");
  console.log(queue);
}

let choice;
do {
  displayMenu();
  choice = parseInt(prompt("Enter your choice: "));

  switch (choice) {
    case 1:
      addElementsToQueue();
      break;
    case 2:
      removeElementFromFront();
      break;
    case 3:
      addAnotherElementToQueue();
      break;
    case 4:
      console.log("Exiting...");
      break;
    default:
      console.log("Invalid choice. Please try again.");
  }
} while (choice !== 4);

File: ProblemSet/Week 2/Problem/Python/Queue.py
queue = []

def display_menu():
  print("Queue Menu:")
  print("1. Add elements to the queue")
  print("2. Remove an element from the front of the queue")
  print("3. Add another element to the queue")
  print("4. Exit")

def add_elements_to_queue():
  num_elements = int(input("Enter the number of elements to add to the queue: "))

  for i in range(num_elements):
    element = input(f"Element {i + 1}: ")
    queue.append(element)

  print("Queue elements:")
  print(queue)

def remove_element_from_front():
  if queue:
    removed_element = queue.pop(0)
    print("Removed element: " + removed_element)
    print("Updated queue elements:")
    print(queue)
  else:
    print("Queue is empty. Cannot remove an element.")

def add_another_element_to_queue():
  new_element = input("Enter the new element: ")
  queue.append(new_element)
  print("Updated queue elements:")
  print(queue)

while True:
  display_menu()
  choice = int(input("Enter your choice: "))

  if choice == 1:
    add_elements_to_queue()
  elif choice == 2:
    remove_element_from_front()
  elif choice == 3:
    add_another_element_to_queue()
  elif choice == 4:
    print("Exiting...")
    break
  else:
    print("Invalid choice. Please try again.")
